test-data/unit/check-functools.test

```diff
@@ -408,33 +408,71 @@ def foo(cls3: Type[B[T]]):
 from typing_extensions import TypedDict, Unpack
 from functools import partial
 
-class Data(TypedDict, total=False):
-    x: int
-
-def f(**kwargs: Unpack[Data]) -> None: ...
-def g(**kwargs: Unpack[Data]) -> None:
-    partial(f, **kwargs)()
-
-class MoreData(TypedDict, total=False):
-    x: int
-    y: int
-
-def f_more(**kwargs: Unpack[MoreData]) -> None: ...
-def g_more(**kwargs: Unpack[MoreData]) -> None:
-    partial(f_more, **kwargs)()
-
-class Good(TypedDict, total=False):
-    y: int
-class Bad(TypedDict, total=False):
-    y: str
-
-def h(**kwargs: Unpack[Data]) -> None:
-    bad: Bad
-    partial(f_more, **kwargs)(**bad)  # E: Argument "y" to "f_more" has incompatible type "str"; expected "int"
-    good: Good
-    partial(f_more, **kwargs)(**good)
```

Comment:
You are deleting this, but it looks like you are not testing a similar situation. All your tests are (effectively) about `partial(f, x=1)(x=2)`, while this one specifically checks for (IMO more realistic/common) use case equivalent to `partial(f, x=1)(y=2)` (non-overlapping arguments that "complete" the call).